conans/client/conan_api.py

```diff
@@ -1495,6 +1495,22 @@ def lock_create(self, path, lockfile_out,
         graph_lock_file.save(lockfile_out)
         self.app.out.info("Generated lockfile: %s" % lockfile_out)
 
+    @api_method
+    def lock_extract_profiles(self, host_profile_name, build_profile_name, lockfile=None, cwd=None):
+        cwd = cwd or os.getcwd()
+        lockfile = lockfile or LOCKFILE
+        lockfile_path = _make_abs_path(lockfile, cwd)
+        host_profile_path = _make_abs_path(host_profile_name, cwd)
+        build_profile_path = _make_abs_path(build_profile_name, cwd)
+        revisions_enabled = self.app.config.revisions_enabled
+        lockfile_handle = GraphLockFile.load(lockfile_path, revisions_enabled)
+        if lockfile_handle.profile_host:
+            save(host_profile_path, lockfile_handle.profile_host.dumps())
+            self.app.out.info("Generated host profile: {}".format(host_profile_path))
+        if lockfile_handle.profile_build:
```

Comment:
We could add a warning message log if build profile is not listed, but by default it's not listed, so I preferred don't alarming.